hive/agents/qnets/sequence_models.py

```diff
@@ -0,0 +1,145 @@
+import torch
+from torch import nn
+
+from hive.utils.registry import registry, Registrable
+from hive.agents.qnets.base import FunctionApproximator
+
+
+class SequenceFn(Registrable):
+    """A wrapper for callables that produce sequence functions."""
+
+    @classmethod
+    def type_name(cls):
+        return "sequence_fn"
+
+
+class SequenceModel(nn.Module):
+    """
+    Base sequence neural network architecture.
+    """
+
+    def __init__(
+        self,
+        rnn_hidden_size=128,
+        num_rnn_layers=1,
+        device="cpu",
```

Comment:
device is an existing param of nn.Module? do we need to redefine?

Reply:
It would look messy if every time we set the device of the hidden state the same one as that of the base network. We chose the easy way that specifies the device of sequence model in the `.yml` config.

Reply:
I dont understand. Isnt the hidden state part of the network module itself: the rnn layers etc.. and when you give the hidden state during forward pass you can use the device arg same as how we handle observation inputs

Reply:
This is fixed by the latest commit, which update the device attribute of the sequence model once we move the whole network to cuda.